src/transformers/modeling_utils.py

```diff
@@ -2321,6 +2334,31 @@ def from_pretrained(
 
         # Load config if we don't provide a configuration
         if not isinstance(config, PretrainedConfig):
+            # In case users load models without auto mapping
+            if is_peft_available() and peft_adapter_model_id is None:
+                raw_adapter_config_dict_path = cls._check_and_return_if_adapter_model(
+                    pretrained_model_name_or_path,
+                    revision=revision,
+                    use_auth_token=use_auth_token,
+                )
+
+                if raw_adapter_config_dict_path is not None:
+                    raw_adapter_config_dict = json.load(open(raw_adapter_config_dict_path, "r"))
```

Comment:
The open needs to be done in a context manager. Here if there is an error while decoding the JSON, the file will stay open.